src/core/DrumSynth.cpp

```diff
@@ -149,34 +148,44 @@ void DrumSynth::GetEnv(int env, const char *sec, const char *key, QString ini)
 
 float DrumSynth::waveform(float ph, int form)
 {
-  float w;
-
-  switch (form)
-  {
-     case 0: w = (float)sin(fmod(ph,TwoPi));                            break; //sine
-     case 1: w = (float)fabs(2.0f*(float)sin(fmod(0.5f*ph,TwoPi)))-1.f; break; //sine^2
-     case 2: while(ph<TwoPi) ph+=TwoPi;
-             w = 0.6366197f * (float)fmod(ph,TwoPi) - 1.f;                     //tri
-             if(w>1.f) w=2.f-w;
-             break;
-     case 3: w = ph - TwoPi * (float)(int)(ph / TwoPi);                        //saw
-             w = (0.3183098f * w) - 1.f;                                break;
-    default: w = (sin(fmod(ph,TwoPi))>0.0)? 1.f: -1.f;                  break; //square
-  }
-
-  return w;
+	float w = 0.0f;
+
+	switch (form)
+	{
+	case 0:
+		w = static_cast<float>(sin(fmod(ph, TwoPi)));
+		break; // sine
+	case 1:
+		w = static_cast<float>(fabs(2.0f * static_cast<float>(sin(fmod(0.5f * ph, TwoPi))) - 1.f));
+		break; // sine^2
+	case 2:
+		while (ph < TwoPi)
+		{
+			ph += TwoPi;
+		}
```

Comment:
This got expanded out again
```suggestion
		while (ph < TwoPi) { ph += TwoPi; }
```